standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/security/TestTokenStoreDelegationTokenSecretManager.java

```diff
@@ -99,19 +112,46 @@ private DelegationTokenIdentifier getID(String tokenStr) throws IOException {
 
   @Test public void testExpiry() throws IOException, InterruptedException {
     DelegationTokenStore tokenStore = new MemoryTokenStore();
-    TokenStoreDelegationTokenSecretManager mgr = createTokenMgr(tokenStore, 1);
+    TokenStoreDelegationTokenSecretManager mgr = createTokenMgr(tokenStore, 1, 8, 2);
     try {
       mgr.startThreads();
       String tokenStr =
           mgr.getDelegationToken(UserGroupInformation.getCurrentUser().getShortUserName(),
               UserGroupInformation.getCurrentUser().getShortUserName());
       DelegationTokenIdentifier id = getID(tokenStr);
       Assert.assertNotNull(mgr.verifyDelegationToken(tokenStr));
-      // Sleep for the renewal duration
-      Thread.sleep(1000);
-      SecretManager.InvalidToken ex = Assert.assertThrows(SecretManager.InvalidToken.class,
+      // Sleep for the expiration/maxlife duration duration
+      Thread.sleep(6000);
+      IllegalStateException ex = Assert.assertThrows(IllegalStateException.class,
```

Comment:
Removal thread not kickin why it will throw exception

Reply:
done